README.md

```diff
@@ -4,57 +4,120 @@ This is a tiny little shim around Cosmos SDK binaries that use the upgrade
-to retrieve it from
-* place the binary for each upgrade at `$DAEMON_HOME/upgrade_manager/<name>`
-where `<name>` is the URI-encoded name of the upgrade as specified in the upgrade
-module plan
-* or, store an os/architecture -> binary URI map in the upgrade plan info field
-as JSON (or YAML??) under the `"binaries"` key, eg:
-```json
-{
-  "binaries": {
-    "linux/amd64":"https://example.com/gaiad?checksum=sha256:b7d96c89d09d9e204f5fedc4d5d55b21"
-  }
-}
+* `DAEMON_NAME` is the name of the binary itself (eg. `xrnd`, `gaiad`)
+* `DAEMON_DOWNLOAD` (optional) if set to `on` will enable auto-downloading of new binaries
+(for security reasons, this is intended for fullnodes rather than validators)
+
+## Folder Layout
+
+`$DAEMON_HOME/upgrade_manager` is expected to belong completely to the upgrade manager and subprocesses
+constrolled by it. Under this folder, we will see the following:
+
 ```
-* or, set the upgrade plan to URI that points to a YAML or JSON file with the above structure 
-that can be retrieved by [go-getter](https://github.com/hashicorp/go-getter) 
-* all arguments passed to the upgrade manager command will be passed to the
-current daemon binary
-* the upgrade manager will monitor the stdout of the daemon to look for signals
-from the upgrade module indicating a pending or required upgrade and act
-appropriately, preferring on-disk binaries when they are present and defaulting
-to binaries provided in the on-chain upgrade plan when they are not
+- bin
+  - upgrade_manager (this binary)
+- genesis
+  - bin
+    - $DAEMON_NAME
+- upgrades
+  - <name>
+    - bin
+      - $DAEMON_NAME
+- current -> upgrades/foo
+```
+
+Each version of the chain is stored under either `genesis` or `upgrades/<name>`, which holds `bin/$DAEMON_NAME`
+along with any other needed files (maybe the cli client? maybe some dlls?). `current` is a symlink to the currently
+active folder (so `current/bin/$DAEMON_NAME` is the binary)
 
+Note: the `<name>` after `upgrades` is the URI-encoded name of the upgrade as specified in the upgrade module plan.
 
-## Layout of `DAEMON_HOME` Folder
+**Question** where is the state of the blockchain (config, data) stored? Under the update manager, or elsewhere?
 
-* `$DAEMON_HOME/upgrade_manager/genesis` or `$DAEMON_HOME/upgrade_manager/genesis/bin/$DAEMON_NAME` should point to the
-genesis binary
-* `$DAEMON_HOME/upgrade_manager/upgrades/<upgrade-name>` or
-`$DAEMON_HOME/upgrade_manager/upgrades/<upgrade-name>/bin/$DAEMON_NAME` should point to the binary for the upgrade
-named `<upgrade-name>`
-* `$DAEMON_HOME/upgrade_manager/current` is symlink'ed to the current binary
+## Usage
 
+Basic Usage:
+
+* The admin is responsible for installing the `upgrade_manager` and setting it as a eg. systemd service to auto-restart
+* The admin is responsible for installing the `genesis` folder manually
+* The admin is responsible for setting the `current` link to point to genesis
```

Comment:
I think it's fine if the `upgrade_manager` sets up current to point to genesis if `current` doesn't exist. I think that's the current behavior and we should leave it as such. Doing otherwise creates complications like sym link vs hard link

Reply:
:+1: